.github/workflows/c2d.yml

```diff
@@ -0,0 +1,67 @@
+##
+## Copyright 2023 Ocean Protocol Foundation
+## SPDX-License-Identifier: Apache-2.0
+##
+name: Provider tests for C2D
+
+on:
+  - push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Provider
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - uses: actions/checkout@v2
+        name: Checkout Barge
+        with:
+          repository: "oceanprotocol/barge"
+          path: 'barge'
+      - name: Login to Docker Hub
+        if: ${{ env.DOCKERHUB_PASSWORD && env.DOCKERHUB_USERNAME }}
+        run: |
+          echo "Login to Docker Hub";echo "$DOCKERHUB_PASSWORD" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
+        env:
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Run Barge
+        working-directory: ${{ github.workspace }}/barge
+        run: |
+          bash -x start_ocean.sh --no-dashboard 2>&1 --with-provider2 --with-c2d > start_ocean.log &
+      - name: Wait for contracts deployment and C2D cluster to be ready
+        working-directory: ${{ github.workspace }}/barge
+        run: |
+          for i in $(seq 1 250); do
+            sleep 10
+            [ -f "$HOME/.ocean/ocean-contracts/artifacts/ready" -a -f "$HOME/.ocean/ocean-c2d/ready" ] && break
+            done
+      - name: Verify deployments
+        run: |
+          netstat -lnpt
+          nc -zv 172.15.0.13 31000
+          cat $HOME/.ocean/ocean-contracts/artifacts/address.json
+          curl http://172.15.0.13:31000
+          nc -zv 172.15.0.16 8080
+      - name: Install dependencies
+        working-directory: ${{ github.workspace }}
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements_dev.txt
+      - name: Test with pytest
+        run: |
+          pytest tests/test_compute.py
+          pytest ocean_provider/utils/test/test_compute.py
+          pytest ocean_provider/utils/test/test_provider_fees.py
+          pytest ocean_provider/validation/test/test_algo_validation.py
+      - name:  docker logs
+        run: docker logs ocean_aquarius_1 && docker logs ocean_provider_1 && docker logs ocean_provider2_1 && docker logs ocean_ipfs_1
+        if: ${{ failure() }}
+      - name: Publish code coverage
```

Comment:
so now we are only reporting coverage from the c2d part? That is not correct. Coverage reports and badges are very important. What coverage do we get just by running c2d tests? That reflects poorly on our code.

Reply:
Yes, it is a possible way to combine the results for coverage report https://docs.knapsackpro.com/2019/how-to-merge-codeclimate-reports-for-parallel-jobs-ci-nodes. I'll try that approach.
cc @alexcos20